pkg/reconciler/v1alpha1/eventlistener/eventlistener.go

```diff
@@ -298,14 +319,32 @@ func (r *Reconciler) reconcileDeployment(logger *zap.SugaredLogger, el *v1alpha1
 			Name:      "config-logging",
 			MountPath: "/etc/config-logging",
 		}},
-		Env: []corev1.EnvVar{{
-			Name: "SYSTEM_NAMESPACE",
-			ValueFrom: &corev1.EnvVarSource{
-				FieldRef: &corev1.ObjectFieldSelector{
-					FieldPath: "metadata.namespace",
-				},
-			},
-		}},
+		Env: env,
+	}
+	var (
+		tolerations                          []corev1.Toleration
+		nodeSelector, annotations, podlabels map[string]string
+		serviceAccountName                   string
+	)
+	podlabels = mergeMaps(el.Labels, GenerateResourceLabels(el.Name))
+
+	// For backward compatibility with podTemplate, serviceAccountName field as part of eventlistener.
+	tolerations = el.Spec.PodTemplate.Tolerations
+	nodeSelector = el.Spec.PodTemplate.NodeSelector
+	serviceAccountName = el.Spec.ServiceAccountName
+
+	if el.Spec.Resources.KubernetesResource != nil {
+		if len(el.Spec.Resources.KubernetesResource.Template.Spec.Tolerations) != 0 {
+			tolerations = el.Spec.Resources.KubernetesResource.Template.Spec.Tolerations
+		}
+		if len(el.Spec.Resources.KubernetesResource.Template.Spec.NodeSelector) != 0 {
+			nodeSelector = el.Spec.Resources.KubernetesResource.Template.Spec.NodeSelector
+		}
+		if el.Spec.Resources.KubernetesResource.Template.Spec.ServiceAccountName != "" {
+			serviceAccountName = el.Spec.Resources.KubernetesResource.Template.Spec.ServiceAccountName
+		}
+		annotations = el.Spec.Resources.KubernetesResource.Template.Annotations
```

Comment:
Don't we also have to mergeMaps for annotations too like we do in 7622d51

Reply:
I feel we don't require here because these are part of `deployment.spec.template.metadata` and there won't be any annotations which are added by k8s, So we don't get continuous loop issue. 

So `deployment.spec.template.metadata.annotation` will add/update to annotations value if provided as part of `el.Spec.Resources.KubernetesResource.Template.Annotations` 

even if user manually edit those directly on deployment it will not allow same like others.

Let me know if you have some other point to address

Reply:
Hmm, yeah you are right...I was confusing `deployment.spec.template.metadata.annotations` with `deployment.metadata.annotations`. I think this is fine then 👍 

Kind of off topic, the reconciler code is getting kind of long and unwieldy, we might want to see how we can refactor/clean it up a bit. I'll open an issue.